### ISSUE TITLE
Refactor coneDir to cardir

### ISSUE THREAD

**Issue by pet1330** [strands-project/strands_qsr_lib#129]
coneDir needs to be refactored. @cdondrup found the following which refers to it as "cardinal directions" 
```
Andrew, U. F. "Qualitative spatial reasoning about cardinal directions." Proc. of the 7th Austrian Conf. on Artificial Intelligence. Baltimore: Morgan Kaufmann. 1991.
```

Maybe **cardir**?

**Comment by cdondrup** [comment]
I'm not sure if it really "needs" to be refactored but I think we should at least list all the QSRs together with the papers where they are described (if any) maybe in the readme or a wiki page. Of course refactoring would make the connection clearer...

**Comment by yianni** [comment]
[See here](http://journals.cambridge.org/download.php?file=%2FKER%2FKER30_01%2FS0269888913000350a.pdf&code=f0b140e5e01f42f5bb451e4f5abcf862)

**Comment by yianni** [comment]
I leave it with you. Have a look on the paper I have given above also as it is mentioned there also (also mentioned as cone directions) but there must be some minimal difference between the two.

**Comment by pet1330** [comment]
I'll have a look on Sunday :smile:

**Comment by pet1330** [comment]
I looked at both of the above papers, the second paper(2013) referenced the first(1991), so I think this should be used. It is also an older paper.

From section 4 of [Qualitative spatial reasoning about cardinal directions (PDF)](http://mapcontext.com/autocarto/proceedings/auto-carto-10/pdf/qualitative-spatial-reasoning-about-cardinal-directions.pdf) says:

>The most often used, prototypical concept of cardinal directions is related to the angular direction between the observer's position and a destination point. This direction is rounded to the next established cardinal direction. The compass is usually divided into 4 major cardinal directions, often with subdivisions for a total of 8 or more directions. This results in cone shaped areas for which a symbolic direction is applicable. We limit the investigation here to the case of 4 and 8 directions. This model of cardinal direction has the property that 'the area of acceptance for any given direction increases with distance1 (Peuquet and Zhan 1987, p. 66) (with additional references) and is sometimes called 'triangular'.

This suggests that the names are reasonably interchangeable, but cardinal directions refers more to the [N E S W] and [N NE E SE S SW W NS] which we have implimented. Cone direction would refer to different number of divided sections. My personal opinion is that cone direction should be a base class which allows the user to specify the size of the division and rotation from centre North. Cardinal direction can then inherit from that, but with specific configurations (i.e. 4 and 8)

**Comment by yianni** [comment]
+1 for renaming to carDir, or maybe better cardDir; blah both sound weird.
@pet1330 could you do it?

**Comment by pet1330** [comment]
Yeah, if everyone is ok with this, I'll sort it out :smile:

**Comment by cdondrup** [comment]
![care](https://cloud.githubusercontent.com/assets/4293484/9716469/d1493d32-5561-11e5-8270-db0750590688.jpg)

:-p

**Comment by yianni** [comment]
Don't you mean?
![image](https://cloud.githubusercontent.com/assets/167472/9717396/535050f6-556a-11e5-9f45-f9c0ea0aa10e.png)

**Comment by cdondrup** [comment]
Equally true ;)

**Comment by yianni** [comment]
closed by #171
